lmfit/minimizer.py

```diff
@@ -950,19 +956,109 @@ def scalar_minimize(self, method='Nelder-Mead', params=None, **kws):
 
         return result
 
+    def _lnprob(self, theta, userfcn, params, var_names, bounds, userargs=(),
+                userkws=None, float_behavior='posterior', is_weighted=True,
+                nan_policy='raise'):
+        """Calculate the log-posterior probability.
+        See the `Minimizer.emcee` method for more details.
+
+        Parameters
+        ----------
+        theta : sequence
+            Float parameter values (only those being varied).
+        userfcn : callable
+            User objective function.
+        params : :class:`~lmfit.parameters.Parameters`
+            The entire set of Parameters.
+        var_names : list
+            The names of the parameters that are varying.
+        bounds : numpy.ndarray
+            Lower and upper bounds of parameters. Has shape (nvarys, 2).
+        userargs : tuple, optional
+            Extra positional arguments required for user objective function.
+        userkws : dict, optional
+            Extra keyword arguments required for user objective function.
+        float_behavior : str, optional
+            Specifies meaning of objective when it returns a float. One of:
+             'posterior' - objective function returnins a log-posterior
+                          probability
+            'chi2' - objective function returns a chi2 value
+        is_weighted : bool
+            If `userfcn` returns a vector of residuals then `is_weighted`
+            specifies if the residuals have been weighted by data uncertainties.
+        nan_policy : str, optional
+            Specifies action if `userfcn` returns NaN values. One of:
+                 'raise' - a `ValueError` is raised
+                'propagate' - the values returned from `userfcn` are un-altered
+                'omit' - the non-finite values are filtered
```

Comment:
I know we currently don't show this function in the documentation so it probably doesn't really matter, but for a list to show up correctly, one would need a newline before/after the elements. Also "raise" is currently not aligned with the other elements